src/deploy/functions/params.ts

```diff
@@ -267,83 +396,271 @@ export async function resolveParams(
+        secretParam.label || secretParam.name
+      }:`,
+    });
+    await secretManager.createSecret(projectId, secretParam.name, {});
+    await secretManager.addVersion(projectId, secretParam.name, secretValue);
+    return secretValue;
+  } else if (!metadata.secretVersion) {
+    // the secret exists, but version "latest" doesnt, which...is questionably possible?
+  }
+
+  // we need to test if the hosting service account can actually read
+  // secretManager.ensureServiceAgentRole(metadata.secret, [`what address goes here`], "roles/secretmanager.admin");
+
+  return secretManager.accessSecretVersion(projectId, secretParam.name, "latest");
+}
+
+async function getSecretMetadata(
+  projectId: string,
+  secretName: string,
+  version: string
+): Promise<{
+  secret?: secretManager.Secret;
+  secretVersion?: secretManager.SecretVersion;
+}> {
+  const secretInfo: any = {};
+  try {
+    secretInfo.secret = await secretManager.getSecret(projectId, secretName);
+    secretInfo.secretVersion = await secretManager.getSecretVersion(projectId, secretName, version);
+  } catch (err: any) {
+    // Throw anything other than the expected 404 errors.
+    if (err.status !== 404) {
+      throw err;
+    }
+  }
+  return secretInfo;
+}
+
 /**
  * Returns the resolved value of a user-defined Functions parameter.
  * Functions params are defined by the output of the Functions SDK, but their value is not set until deploy-time.
  *
  * For most param types, we check the contents of the dotenv files first for a matching key, then interactively prompt the user.
  * When the CLI is running in non-interactive mode or with the --force argument, it is an error for a param to be undefined in dotenvs.
  */
-async function promptParam(param: Param, resolvedDefault?: ParamValue): Promise<ParamValue> {
+async function promptParam(
+  param: Param,
+  projectId: string,
+  resolvedDefault?: RawParamValue
+): Promise<ParamValue> {
   if (param.type === "string") {
-    return promptStringParam(param, resolvedDefault as string | undefined);
+    const provided = await promptStringParam(
+      param,
+      projectId,
+      resolvedDefault as string | undefined
+    );
+    return new ParamValue(provided.toString(), false, { string: true });
   } else if (param.type === "int") {
-    return promptIntParam(param, resolvedDefault as number | undefined);
+    const provided = await promptIntParam(param, resolvedDefault as number | undefined);
+    return new ParamValue(provided.toString(), false, { number: true });
+  } else if (param.type === "boolean") {
+    const provided = await promptBooleanParam(param, resolvedDefault as boolean | undefined);
+    return new ParamValue(provided.toString(), false, { boolean: true });
+  } else if (param.type === "secret") {
+    throw new FirebaseError(
+      `Somehow ended up trying to interactively prompt for secret parameter ${param.name}, which should never happen.`
```

Comment:
I'm a little confused why `handleSecret` implements prompting for secret params instead of doing it here (where I'd naturally look for prompting logic).

Reply:
Still not sure - is there reason why secretParam given super special treatment?